PSReadLine/DisplayBlockBase.cs

```diff
@@ -0,0 +1,79 @@
+/********************************************************************++
+Copyright (c) Microsoft Corporation.  All rights reserved.
+--********************************************************************/
+
+using System;
+using Microsoft.PowerShell.Internal;
+
+namespace Microsoft.PowerShell
+{
+    public partial class PSConsoleReadLine
+    {
+        private class DisplayBlockBase
+        {
+            internal PSConsoleReadLine Singleton;
+            internal int Top;
+            //internal int PreviousTop;
+            //internal int ColumnWidth;
+            //internal int Rows;
+            //internal int Columns;
+
+            private protected void MoveCursorDown(int cnt)
+            {
+                IConsole console = Singleton._console;
+                while (cnt-- > 0)
+                {
+                    console.Write("\n");
+                }
+            }
+
+            private protected void AdjustForPossibleScroll(int cnt)
+            {
+                IConsole console = Singleton._console;
+                var scrollCnt = console.CursorTop + cnt + 1 - console.BufferHeight;
+                if (scrollCnt > 0)
+                {
+                    Top -= scrollCnt;
+                    _singleton._initialY -= scrollCnt;
+                    _savedCursorTop -= scrollCnt;
+                }
+            }
+
+            private int _savedCursorLeft;
+            private int _savedCursorTop;
+
+            public void SaveCursor()
+            {
+                IConsole console = Singleton._console;
+                _savedCursorLeft = console.CursorLeft;
+                _savedCursorTop = console.CursorTop;
+            }
+
+            public void RestoreCursor() => Singleton._console.SetCursorPosition(_savedCursorLeft, _savedCursorTop);
+
+            private protected void WriteBlankLines(int top, int count)
```

Comment:
`PSConsoleReadLine` has an implementation of `WriteBlankLines(int top, int count)`. Please use `_singleton` to access it instead of have a duplicate implementation.

Reply:
Removed from here.